packages/clerk-js/src/ui/elements/PhoneInput/PhoneInput.tsx

```diff
@@ -37,9 +42,48 @@ export const PhoneInput = (props: PhoneInputProps) => {
   const selectedCountryOption = React.useMemo(() => {
     return countryOptions.find(o => o.country.iso === selectedIso) || countryOptions[0];
   }, [selectedIso]);
+  const dynamicPadding = selectedCountryOption.country.code.length * 5; // this to calculate the padding of the input field depending the length of country code
 
   React.useEffect(callOnChangeProp, [cleanPhoneNumber]);
 
+  const handlePaste = (e: ClipboardEvent<HTMLInputElement>) => {
+    const inputValue = e.clipboardData.getData('text');
+    const testPattern = /\+/g;
+
+    if (testPattern.test(inputValue)) {
+      setHasBeenPasted(true);
```

Comment:
If I understand correctly, you use this state to avoid handling the change twice and thus duplicating the input, correct?

Reply:
@nikosdouvlis if I paste a value it will also trigger the onChange resulting in an unexpected state change. For example, the +123123123123(US code) would result in (+1) in select, (231)231 2312312. The last two digits(12 in our case) was added from the onChange. Hope this makes it clearer.

Reply:
Is it possible to handle the value parsing and setting in the onChange event or does that create a clunky experience?

Reply:
Can we remove the explicitly set state by using `e.preventDefault()` as shown [here](https://github.com/clerkinc/javascript/blob/2bbb13022e4a9d3cbd62f06ad2b06b229f05583e/packages/clerk-js/src/ui/elements/CodeControl.tsx#L114-L113)?

Reply:
@gkats any updates on this?